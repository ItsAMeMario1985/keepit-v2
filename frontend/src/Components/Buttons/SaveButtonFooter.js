import styled from 'styled-components/macro'
import { ReactComponent as SaveFooterButton } from '../../Assets/save-footer-button.svg'

export default function SaveButtonFooter({ onClick }) {
  return (
    <StyledIcon>
      <SaveFooterButton onClick={onClick} height="50px"></SaveFooterButton>
    </StyledIcon>
  )
}

const StyledIcon = styled.div`
  height: 60px;
  margin: auto;
  margin-top: 6px;
`
