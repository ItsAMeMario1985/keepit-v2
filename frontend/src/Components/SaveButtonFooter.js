import ImageUploading from 'react-images-uploading'
import { useHistory, useLocation } from 'react-router-dom'
import { ReactComponent as ActionButtonSvg } from '../Assets/keepit-button.svg'
import styled from 'styled-components/macro'
import { ReactComponent as SaveFooterButton } from '../Assets/save-footer-button.svg'

export default function SaveButtonFooter({ onClick }) {
  const history = useHistory()
  const maxNumber = 3
  const location = useLocation()

  const onChange = (imageList, addUpdateIndex) => {
    history.push('/new', { images: imageList })
    if (location.pathname === '/new') {
      window.location.reload()
    }
  }

  return (
    <div>
      <ImageUploading
        multiple
        onChange={onChange}
        maxNumber={maxNumber}
        dataURLKey="data_url"
      >
        {({ onImageUpload }) => (
          <div>
            <StyledIcon>
              <SaveFooterButton
                onClick={onClick}
                height="50px"
              ></SaveFooterButton>
            </StyledIcon>
          </div>
        )}
      </ImageUploading>
    </div>
  )
}

const StyledIcon = styled.div`
  height: 60px;
  margin: auto;
  margin-top: 6px;
`
