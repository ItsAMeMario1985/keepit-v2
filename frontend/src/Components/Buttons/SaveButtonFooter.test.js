import { render } from '@testing-library/react'
import SaveButtonFooter from './SaveButtonFooter'
describe('SaveButtonFooter', () => {
  it('renders correctly', () => {
    const { container } = render(<SaveButtonFooter />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
