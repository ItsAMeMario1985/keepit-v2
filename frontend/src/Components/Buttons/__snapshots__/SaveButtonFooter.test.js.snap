// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveButtonFooter renders correctly 1`] = `
<div
  class="SaveButtonFooter__StyledIcon-r9ejj3-0 hLypKd"
>
  <svg
    height="50px"
  >
    save-footer-button.svg
  </svg>
</div>
`;
